src/shims/time.rs

```diff
@@ -154,46 +164,73 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         // This may not be consistent with libc::localtime_r's result.
         let tm_isdst = -1;
 
-        // tm_zone represents the timezone value in the form of: +0730, +08, -0730 or -08.
-        // This may not be consistent with libc::localtime_r's result.
-        let offset_in_seconds = dt.offset().fix().local_minus_utc();
-        let tm_gmtoff = offset_in_seconds;
-        let mut tm_zone = String::new();
-        if offset_in_seconds < 0 {
-            tm_zone.push('-');
-        } else {
-            tm_zone.push('+');
-        }
-        let offset_hour = offset_in_seconds.abs() / 3600;
-        write!(tm_zone, "{:02}", offset_hour).unwrap();
-        let offset_min = (offset_in_seconds.abs() % 3600) / 60;
-        if offset_min != 0 {
-            write!(tm_zone, "{:02}", offset_min).unwrap();
+        match this.tcx.sess.target.os.as_ref() {
+            "solaris" | "illumos" => {
+                // solaris/illumos system tm struct does not have
+                // the additional tm_zone/tm_gmtoff fields.
```

Comment:
instead of duplicating the core part, can you skip the parts that write the extra fields?